python/ray/_private/worker.py

```diff
@@ -846,6 +851,88 @@ def get_gpu_ids():
     Returns:
         A list of GPU IDs.
     """
+    assigned_ids = _get_all_resource_ids("GPU", r"^GPU_group_[0-9A-Za-z]+$")
+    # If the user had already set CUDA_VISIBLE_DEVICES, then respect that (in
+    # the sense that only GPU IDs that appear in CUDA_VISIBLE_DEVICES should be
+    # returned).
+    if global_worker.original_gpu_ids is not None:
+        assigned_ids = [
+            global_worker.original_gpu_ids[gpu_id] for gpu_id in assigned_ids
+        ]
+        # Give all GPUs in local_mode.
+        if global_worker.mode == LOCAL_MODE:
+            max_gpus = global_worker.node.get_resource_spec().num_gpus
+            assigned_ids = global_worker.original_gpu_ids[:max_gpus]
+
+    return assigned_ids
+
+
+@DeveloperAPI()
+@client_mode_hook
+def get_neuron_core_ids() -> List[str]:
```

Comment:
chatted with the team, instead of exposing 
`ray.get_neuron_core_ids`, we could have a function in `ray.get_runtime_context().get_resources_ids()` which returns a map from resource name to ids, so user can use it this way

```
resource_ids = ray.get_runtime_context().get_resources_ids()
# Use resource_ids["neuron_core"]
```